splitio/storage/pluggable.py

```diff
@@ -470,3 +472,87 @@ def put(self, segment):
 #        except Exception:
 #            _LOGGER.error('Error updating segment storage')
 #            _LOGGER.debug('Error: ', exc_info=True)
+
+
+class PluggableImpressionsStorage(ImpressionStorage):
+
+    def __init__(self, pluggable_adapter, sdk_metadata, prefix=None):
+        """
+        Class constructor.
+
+        :param pluggable_adapter: Storage client or compliant interface.
+        :type pluggable_adapter: TBD
+        :param sdk_metadata: SDK & Machine information.
+        :type sdk_metadata: splitio.client.util.SdkMetadata
+        """
+        self._pluggable_adapter = pluggable_adapter
+        self._sdk_metadata = sdk_metadata
+        self._impressions_queue_key = 'SPLITIO.impressions'
+        if prefix is not None:
+            self._impressions_queue_key = prefix + "." + self._impressions_queue_key
+
+    def _wrap_impressions(self, impressions):
+        """
+        Wrap impressions to be stored in storage
+
+        :param impressions: Impression to add to the queue.
+        :type impressions: splitio.models.impressions.Impression
+
+        :return: Processed impressions.
+        :rtype: list[splitio.models.impressions.Impression]
+        """
+        bulk_impressions = []
+        for impression in impressions:
+            if isinstance(impression, Impression):
+                to_store = {
+                    'm': {  # METADATA PORTION
```

Comment:
Since `self._sdk_metadata` is an immutable object, we can create the `'m'` object only once in the `__init__` function.

For example, instead of doing `self._sdk_metadata = sdk_metadata` we can do something like 
```
self._sdk_metadata = {  # METADATA PORTION
                        's': sdk_metadata.sdk_version,
                        'n': sdk_metadata.instance_name,
                        'i': sdk_metadata.instance_ip,
}
```
and then assign to `m` the object: `'m': self._sdk_metadata`

Reply:
will do